### DISCUSSION TITLE
Generating NDepend Report from the Uno CI/CD Pipeline

### DISCUSSION THREAD

**Discussion by ndepend** [unoplatform/uno#16943] [General]
Hello,

As discussed today, it would be beneficial to generate an NDepend Report from the Uno CI/CD pipeline. This initiative aims to provide developers and contributors with insights into the code and security enhancements that have been achieved and those that are pending.

You can view some sample reports at these links:
- [NDepend Sample Reports](https://www.ndepend.com/sample-reports/)
- [On Orchard Core 1.8 vs 1.7](https://www.ndepend.com/Sample-Reports/OnOrchadCore/NDependReport.html#issues-summary)
- [Source Level View](https://www.ndepend.com/Sample-Reports/OnOrchadCore/NDependReportFiles/src/ContentDefinitionManager.cs.html)

To initiate this process, please provide the branch you would like NDepend to analyze first, along with instructions on how to compile it. This will enable NDepend to generate an initial report.

Subsequently, we will discuss how to establish the baseline, import code coverage data from tests, and collect issues identified by Roslyn Analyzers. These steps are summarized in the following diagram: [NDepend Analysis Input](https://www.ndepend.com/Doc/CI_Inputs/NDependAnalysisInput.png).

Thank you.

**Comment by jeromelaban** [comment]
@ndepend Thanks! Let's start with [`release/stable/5.2`](https://github.com/unoplatform/uno/tree/release/stable/5.2). The 

You'll need to:
- Create a copy of the `src/crosstargeting_override.props.sample` file to `src/crosstargeting_override.props`, then set `UnoTargetFrameworkOverride` to `net8.0`.
- Then build the solution filter: `https://github.com/unoplatform/uno/blob/master/src/Uno.UI-Skia-only.slnf`

We'll see for the other target frameworks in a future pass! Let us know if you need something else.
